beacon-chain/sync/pending_attestations_queue.go

```diff
@@ -326,8 +348,20 @@ func (s *Service) validatePendingAtts(ctx context.Context, slot primitives.Slot)
 
 	for bRoot, atts := range s.blkRootToPendingAtts {
 		for i := len(atts) - 1; i >= 0; i-- {
-			if slot >= atts[i].AggregateAttestationAndProof().AggregateVal().GetData().Slot+params.BeaconConfig().SlotsPerEpoch {
-				// Remove the pending attestation from the list in place.
+			var attSlot primitives.Slot
```

Comment:
do we need a check above that atts is at least 1?

Reply:
No, `i` will be initialized to -1 and so the `>= 0` condition will be never met, and therefore the inside of the loop will never be executed.